domain/src/test/java/org/oppia/domain/topic/TopicControllerTest.kt

```diff
@@ -750,13 +776,344 @@ class TopicControllerTest {
+    }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testCompletedStoryList_finishOneEntireStoryAndOneChapterInAnotherStory_completedStoryListIsCorrect() =
+    runBlockingTest(coroutineContext) {
+      markFractionsStory0Chapter0AsCompleted()
+      advanceUntilIdle()
+
+      markRatiosStory0Chapter0AsCompleted()
+      advanceUntilIdle()
+
+      markRatiosStory0Chapter1AsCompleted()
+      advanceUntilIdle()
+
+      topicController.getCompletedStoryList(profileId1).observeForever(mockCompletedStoryListObserver)
+      advanceUntilIdle()
+
+      verifyCompletedStoryListSucceeded()
+
+      val completedStoryList = completedStoryListResultCaptor.value.getOrThrow()
+      assertThat(completedStoryList.storySummaryCount).isEqualTo(1)
+      assertThat(completedStoryList.storySummaryList[0].storyId).isEqualTo(RATIOS_STORY_ID_0)
+    }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testCompletedStoryList_finishTwoStories_completedStoryListIsCorrect() =
+    runBlockingTest(coroutineContext) {
+      markFractionsStory0Chapter0AsCompleted()
+      advanceUntilIdle()
+
+      markFractionsStory0Chapter1AsCompleted()
+      advanceUntilIdle()
+
+      markRatiosStory0Chapter0AsCompleted()
+      advanceUntilIdle()
+
+      markRatiosStory0Chapter1AsCompleted()
+      advanceUntilIdle()
+
+      topicController.getCompletedStoryList(profileId1).observeForever(mockCompletedStoryListObserver)
+      advanceUntilIdle()
+
+      verifyCompletedStoryListSucceeded()
+
+      val completedStoryList = completedStoryListResultCaptor.value.getOrThrow()
+      assertThat(completedStoryList.storySummaryCount).isEqualTo(2)
+      assertThat(completedStoryList.storySummaryList[0].storyId).isEqualTo(FRACTIONS_STORY_ID_0)
+      assertThat(completedStoryList.storySummaryList[1].storyId).isEqualTo(RATIOS_STORY_ID_0)
+    }
+
   private fun setUpTestApplicationComponent() {
     DaggerTopicControllerTest_TestApplicationComponent.builder()
       .setApplication(ApplicationProvider.getApplicationContext())
       .build()
       .inject(this)
   }
 
+  private fun markFractionsStory0Chapter0AsCompleted(){
+    storyProgressController.recordCompletedChapter(
+      profileId1,
+      FRACTIONS_TOPIC_ID,
+      FRACTIONS_STORY_ID_0,
+      FRACTIONS_EXPLORATION_ID_0
+    ).observeForever(mockRecordProgressObserver)
+  }
+
+  private fun markFractionsStory0Chapter1AsCompleted(){
+    storyProgressController.recordCompletedChapter(
+      profileId1,
+      FRACTIONS_TOPIC_ID,
+      FRACTIONS_STORY_ID_0,
+      FRACTIONS_EXPLORATION_ID_1
+    ).observeForever(mockRecordProgressObserver)
+  }
+
+  private fun markRatiosStory0Chapter0AsCompleted(){
+    storyProgressController.recordCompletedChapter(
+      profileId1,
+      RATIOS_TOPIC_ID,
+      RATIOS_STORY_ID_0,
+      RATIOS_EXPLORATION_ID_0
+    ).observeForever(mockRecordProgressObserver)
+  }
+
+  private fun markRatiosStory0Chapter1AsCompleted(){
```

Comment:
Nit on spacing here & above: extra space needed between ``)`` and ``{``.

Reply:
Done, here and everywhere else.